python_modules/dagster/dagster/_core/definitions/graph_definition.py

```diff
@@ -226,7 +226,10 @@ def __init__(
             name=name,
             description=description,
             input_defs=input_defs,
-            output_defs=[output_mapping.definition for output_mapping in self._output_mappings],
+            output_defs=[
+                output_mapping.get_definition(is_dynamic=dynamic_mappings[output_mapping.out_name])
```

Comment:
It's a little awkward to pass in a boolean to determine the type of the returned object. Thoughts on just constructing the `OutputDefinition` / `DynamicOutputDefinition` here with the data from the output mapping?